### PR TITLE
CMake: also update BASE_LINKER_FLAGS* when dropping -fuse-ld=...

### PR DESCRIPTION
**Pull request by tamiko**

### CODE REVIEW

**Comment by tamiko** [comment]
@jppelteret *ping*

**Comment by jppelteret** [comment]
Worked for me. Thanks!

File: cmake/setup_finalize.cmake [hunk 1]
```diff
@@ -99,12 +99,14 @@ FOREACH(build ${DEAL_II_BUILD_TYPES})
...
-      )
-    STRING(REPLACE "${_linker_flag}" "${_replacement_flag}"
-      DEAL_II_LINKER_FLAGS_${build} "${DEAL_II_LINKER_FLAGS_${_build}}"
-      )
+    FOREACH(_flags
+        DEAL_II_LINKER_FLAGS DEAL_II_LINKER_FLAGS_${build}
+        BASE_LINKER_FLAGS BASE_LINKER_FLAGS_${build}
+        )
+      STRING(REPLACE "${_linker_flag}" "${_replacement_flag}"
+        ${_flags} "${${_flags}}"
```
Comment (tjhei) [inline cmake/setup_finalize.cmake]: isn't cmake wonderful? :)

**Pull request merged**